src/store/actions.js

```diff
@@ -91,44 +92,117 @@ const loadCurrentModel = async ({ commit, state }, payload) => {
   } else {
     commit(types.CLEAR_CURRENT_PRODUCT)
-    commit(types.LOADING, { loading: false })
+    commit(types.SET_LOADING, false)
   }
 }
 
-const createNewPolicy = async ({ commit }, imei) => {
-  commit(types.CLEAR_CURRENT_POLICY)
-  commit(types.LOADING, { loading: true })
+const createNewPolicy = async ({ commit, state }, payload) => {
+  // We need custom axios instance to handle 404 differently
+  const customAxios = axios.create({ baseUrl: process.env.API_ADDRESS })
+  const { deviceId, productId } = payload
+  commit(types.SET_LOADING, false)
+  commit(types.CLEAR_POLICY_LOADING_INFO)
+
+  const policyLoadingInfo = {
+    deviceId: deviceId
+  }
+
+  commit(types.SET_POLICY_LOADING_INFO, policyLoadingInfo)
 
   let response = null
 
+  // STEP 1: getting task ID
   try {
-    response = await axios.post('insurance/policy-draft/' + imei)
-  } catch (e) {
-    commit(types.LOADING, { loading: false })
+    response = await customAxios.get('insurance/policy/android/pair/' + deviceId)
+  } catch (error) {
+    var newPolicyLoadingInfo = Object.assign({}, policyLoadingInfo)
+
+    if (error.response.status === 404) {
+      newPolicyLoadingInfo.notFound = true
+    } else if (error.response.status === 503 || error.response.status === 500) {
+      newPolicyLoadingInfo.serverError = true
+    }
+
+    commit(types.SET_POLICY_LOADING_INFO, newPolicyLoadingInfo)
+    return
   }
 
-  if (response && response.data.policy) {
-    commit(types.LOAD_CURRENT_POLICY, response.data)
-    commit(types.LOADING, { loading: false })
+  commit(types.CLEAR_POLICY_LOADING_INFO)
+  policyLoadingInfo.taskId = response.data.taskId
+  commit(types.SET_POLICY_LOADING_INFO, policyLoadingInfo)
+
+  // STEP 2: gettings task
+  let waitMultiplier = 1
+  while (!response.data.policyId && !response.data.validationResultCode && state.policyLoadingInfo.taskId) {
+    response = await axios.post('insurance/policy/android', {
+      TaskId: policyLoadingInfo.taskId,
+      ProductId: productId
+    })
+
+    await sleep(waitMultiplier * 250)
+    waitMultiplier++
+  }
+
+  commit(types.CLEAR_POLICY_LOADING_INFO)
+
+  if (response.data.validationResultCode) {
+    policyLoadingInfo.validationResultCode = response.data.validationResultCode
   } else {
-    commit(types.CLEAR_CURRENT_POLICY)
-    commit(types.LOADING, { loading: false })
+    policyLoadingInfo.policyId = response.data.policyId
   }
+
+  commit(types.SET_POLICY_LOADING_INFO, policyLoadingInfo)
+}
+
+const getPolicy = async ({ commit }, policyId) => {
+  commit(types.SET_LOADING, true)
+  commit(types.CLEAR_CURRENT_POLICY)
+  const response = await axios.get('insurance/policy/android/' + policyId)
+
+  if (response && response.data) {
+    commit(types.SET_CURRENT_POLICY, response.data)
+  }
+
+  commit(types.SET_LOADING, false)
+}
+
+const sendPolicyPayment = async ({ commit, state }) => {
+  const web3 = state.userWeb3.web3Instance()
+  const productAddress = '0x146392ae1d5f8894104a531f41ea62e5efa9bbc3'
```

Comment:
const productAddress = '0x146392ae1d5f8894104a531f41ea62e5efa9bbc3'
this really should be hardcoded?

Reply:
✅ Removed